pennylane/labs/resource_estimation/templates/stateprep.py

```diff
@@ -0,0 +1,358 @@
+    """Resource class for the MottonenStatePreparation template.
+
+    Args:
+        state_vector (tensor_like): Input array of shape ``(2^n,)``, where ``n`` is the number of wires
+            the state preparation acts on. The input array must be normalized.
+        wires (Iterable): wires that the template acts on
+
+    Resource Parameters:
+        * num_wires(int): the number of wires that StatePrep acts on
+
+    Resources:
+        Using the resources as described in `Mottonen et al. (2008) <https://arxiv.org/pdf/quant-ph/0407010>`_.
+        The resources are defined as :math:`2^{N+2} - 5` :class:`~.ResourceRZ` gates and
+        :math:`2^{N+2} - 4N - 4` :class:`~.ResourceCNOT` gates.
+    """
+
+    @staticmethod
+    def _resource_decomp(num_wires, **kwargs) -> Dict[CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources of the operator. The
+        keys are the operators and the associated values are the counts.
+
+        Args:
+            num_wires(int): the number of wires that StatePrep acts on
+
+        Resources:
+            Using the resources as described in `Mottonen et al. (2008) <https://arxiv.org/pdf/quant-ph/0407010>`_.
+            The resources are defined as :math:`2^{N+2} - 5` :class:`~.ResourceRZ` gates and
+            :math:`2^{N+2} - 4N - 4` :class:`~.ResourceCNOT` gates.
+        """
+        gate_types = {}
+        rz = re.ResourceRZ.resource_rep()
+        cnot = re.ResourceCNOT.resource_rep()
+
+        r_count = 2 ** (num_wires + 2) - 5
+        cnot_count = 2 ** (num_wires + 2) - 4 * num_wires - 4
+
+        if r_count:
+            gate_types[rz] = r_count
+
+        if cnot_count:
+            gate_types[cnot] = cnot_count
+        return gate_types
+
+    @property
+    def resource_params(self) -> dict:
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Returns:
+            dict: A dictionary containing the resource parameters:
+                * num_wires(int): the number of wires that StatePrep acts on
+        """
+        return {"num_wires": len(self.wires)}
+
+    @classmethod
+    def resource_rep(cls, num_wires) -> CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the Operator that are needed to compute a resource estimation.
+
+        Args:
+            num_wires(int): the number of wires that StatePrep acts on
+
+        Returns:
+            CompressedResourceOp: the operator in a compressed representation
+        """
+        params = {"num_wires": num_wires}
+        return CompressedResourceOp(cls, params)
+
+    @classmethod
+    def tracking_name(cls, num_wires) -> str:
+        return f"MottonenStatePrep({num_wires})"
+
+
+class ResourceSuperposition(qml.Superposition, ResourceOperator):
+    """Resource class for the `:class:~.Superposition` template.
```

Comment:
See ResourceQFT as an example. Just missing a .. seealso  and a simple example
```suggestion
    """Resource class for the Superposition template.
```